### PR TITLE
chore: add clean-changelog script

### PR DESCRIPTION
**Pull request by lafirest**

### CODE REVIEW

**Comment by zmstone** [comment]
it'd be easier to read like below:

```
is_released() {
  file="$1"
  prefix="$2"
  latest_commit=$(git log -1 --pretty=format:"%H" "$file")
  git tag --contains $latest_commit | grep -q -E "^${prefix}5\.[0-9]+\.[0-9]+$"
}

for $file in ${ce_files[@]}; do
  if is_released "$file" 'v' && is_released "$file" 'e'; then
    git rm $file
  fi
done

for $file in ${ee_files[@]}; do
  if is_released "$file" 'e'; then
    git rm $file
  fi
done
```

I did not test it though.

File: scripts/clean-changelog.sh [hunk 1]
```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
```
Comment (qzhuyan) [inline scripts/clean-changelog.sh:1]: nit, add some README

File: scripts/clean-changelog.sh [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        [ -n "$snd_check_name" ] && [[ ! "$snd_merged" =~ $PR ]] && flag=0
+        if [[ $flag -eq 1 ]]; then
+            echo "deleted $target PR: #$PR"
+            find "$dir" -name "*$PR*.md" -type f -delete
+        fi
+    done
+    git add "$dir"
+}
+
+do_clean "$ee_name" "$ee_name"
```
Comment (qzhuyan) [inline scripts/clean-changelog.sh:60]: question, shall we handle both `ee` and `ce` per script execution?

**Comment by ieQu1** [comment]
I am not sure if we need to automate this. This script should be somewhat aware of our branching strategy and whatnot, so it will add maintenance burden for us. Git keeps all deleted files in the repo forever anyway, so it won't help to speed up cloning of the repo. 

I think it will be sufficient to simply delete files that are "obviously very old" from time to time manually.

**Comment by zmstone** [comment]
> I am not sure if we need to automate this. This script should be somewhat aware of our branching strategy and whatnot, so it will add maintenance burden for us. Git keeps all deleted files in the repo forever anyway, so it won't help to speed up cloning of the repo.
> 
> I think it will be sufficient to simply delete files that are "obviously very old" from time to time manually.

not intended for automation.

**Pull request closed without merging**